### DISCUSSION TITLE
How to sync activity removed from Trakt

### DISCUSSION THREAD

**Discussion by frlefebvre** [trakt/trakt-api#411] [Q&A]
Hi there,

I'm wondering if there's a way to know that a user has removed activity from the Trakt website. For example, a user has first marked a whole season as watched. My app now has all these episodes saved as being watched. Later on, if the user decides to remove that activity (not reset), how can my app know these are no longer considered watched?

The Last Activities watched_at dates seem to reflect when a user removes a watch activity but these don't appear when you Get Watched History. I could call Get Watched every time I sync but this seems overkill for both Trakt and myself.

**Comment by frlefebvre** [comment]
Looking into this some more, here is the only alternative I can think of that would proactively let my app know that some episodes have been "unwatched":
- Query the watched endpoint for shows without season/episodes details.
- For each shows, compare the "last_updated_at" date.
- If different than the one on device, query the progress endpoint for that show.
- Go through every episode and update if necessary.
- Store the new "last_updated_at" date for next time.

Compared to simply requesting the episodes history since the last updated at date, this would result in many more requests being made and much more on-device processing. Not to mention just longer sync times for users. 

Am I missing something? I'd like to be a good API citizen and avoid unnecessary requests but I also need to make sure my app accurately mirrors Trakt's content. The perfect solution would be for "unwatch" to be communicated in the history.

**Reply by rectifyer** [reply]
The logic you have is good, but step 1 should always be checking [/sync/last_activities](https://trakt.docs.apiary.io/#reference/sync/last-activities/get-last-activity). That is a very quick API call and you can save and compare the timestamps to know if you need to take further action. For example, you'll only need to re-sync the `/sync/watched/shows` if the `episodes.watched_at` timestamp has changed in `/sync/last_activities`.
